internal/datastore/postgres/locks.go

```diff
@@ -26,7 +25,7 @@ func (pgd *pgDatastore) tryAcquireLock(ctx context.Context, lockID lockID) (bool
 	// > even if other sessions are awaiting the lock; this statement is true regardless of whether the
 	// > existing lock hold and new request are at session level or transaction level.
 	// See: https://www.postgresql.org/docs/current/explicit-locking.html#ADVISORY-LOCKS
-	row := pgd.writePool.QueryRow(ctx, `
+	row := pgd.gcConn.QueryRow(ctx, `
```

Comment:
Do we only use this lock for GC? Is it a problem if it's used for something else? Are we using the GC connection pool because we assume that it will usually be free?